radix-common/src/crypto/bls12381/signature.rs

```diff
@@ -42,7 +42,35 @@ impl Bls12381G2Signature {
     }
 
     /// Aggregate multiple signatures into a single one
-    pub fn aggregate(signatures: &[Bls12381G2Signature]) -> Result<Self, ParseBlsSignatureError> {
+    /// This method does validate provided input signatures.
+    pub fn aggregate(signatures: &[Self]) -> Result<Self, ParseBlsSignatureError> {
+        if !signatures.is_empty() {
+            let sig_first = signatures[0].to_native_signature()?;
+
+            sig_first.validate(true)?;
+
+            let mut agg_sig = AggregateSignature::from_signature(&sig_first);
```

Comment:
Similar to the `AggregatePublicKey`, I feel we should use `AggregateSignature::aggregate` or better yet, `AggregateSignature::aggregate_serialized`? And perhaps pass through a `sigs_groupcheck: bool` parameter?

Reply:
Done